### PR TITLE
FF90: Adds support for HTTP Sec-Fetch-* headers

### PR DESCRIPTION
**Pull request by hamishwillee**
This updates the documents for `Sec-Fetch-*` headers as requested in #5375 (for https://bugzilla.mozilla.org/show_bug.cgi?id=1695911)

- [x] [Sec-Fetch-Site](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Site)
- [x] [Sec-Fetch-Mode](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Mode)
- [x] [Sec-Fetch-User](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-User)
- [x] [Sec-Fetch-Dest](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Dest)
- [x] Release note

This depends a bit on BCD to tell me if this should still be considered experimental:  [BCD](https://github.com/mdn/browser-compat-data/pull/10679)

### CODE REVIEW

File: files/en-us/web/http/headers/index.html [hunk 1]
```diff
@@ -332,15 +332,17 @@ <h3 id="HTTP_Public_Key_Pinning_GlossaryHPKP">HTTP Public Key Pinning (HPKP)</h3
 
 <h3 id="Fetch_metadata_request_headers">Fetch metadata request headers</h3>
 
+<p>{{Glossary("Fetch metadata request header", "Fetch metadata request headers")}} provides information about the context from which the request originated. This allows a server to make decisions about whether a request should be allowed based on where the request came from and how the resource will be used.</p>
+
 <dl>
  <dt>{{HTTPHeader("Sec-Fetch-Site")}}</dt>
  <dd>It is a request header that indicates the relationship between a request initiator's origin and its target's origin. It is a Structured Header whose value is a token with possible values <code>cross-site</code>, <code>same-origin</code>, <code>same-site</code>, and <code>none</code>.</dd>
  <dt>{{HTTPHeader("Sec-Fetch-Mode")}}</dt>
- <dd>It is a request header that indicates the request's mode to a server. It is a Structured Header whose value is a token with possible values <code>cors</code>, <code>navigate</code>, <code>nested-navigate</code>, <code>no-cors</code>, <code>same-origin</code>, and <code>websocket</code>.</dd>
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/index.html:339]: Note, `nested-navigate` and `nested-document` are not present in the spec

File: files/en-us/web/http/headers/sec-fetch-dest/index.html [hunk 2]
```diff
@@ -19,26 +25,27 @@
...
 <h2 id="Syntax">Syntax</h2>
 
-<pre class="brush: html">Sec-Fetch-Dest: audio
+<pre class="brush: http">Sec-Fetch-Dest: audio
 Sec-Fetch-Dest: audioworklet
 Sec-Fetch-Dest: document
 Sec-Fetch-Dest: embed
 Sec-Fetch-Dest: empty
 Sec-Fetch-Dest: font
+Sec-Fetch-Dest: frame
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/sec-fetch-dest/index.html:45]: note, `frame` and `iframe` have an issue note in the spec "frame and iframe haven’t landed in Fetch yet. <https://github.com/whatwg/fetch/issues/948>". But that PR has merged, so I believe these should be included.

File: files/en-us/glossary/fetch_metadata_request_header/index.html [hunk 3]
```diff
@@ -5,24 +5,29 @@
...
+<p>These headers are prefixed with <code>Sec-</code>, and hence have {{Glossary("Forbidden header name", "forbidden header names")}}. As such, they cannot be modified from JavaScript.</p>
+
+<p>The fetch metadata request headers are:</p>
 
 <ul>
  <li>{{HTTPHeader("Sec-Fetch-Site")}}</li>
  <li>{{HTTPHeader("Sec-Fetch-Mode")}}</li>
  <li>{{HTTPHeader("Sec-Fetch-User")}}</li>
  <li>{{HTTPHeader("Sec-Fetch-Dest")}}</li>
+ <li><a href="https://web.dev/fetch-metadata/">Protect your resources from web attacks with Fetch Metadata</a> (web.dev)</li>
```
Comment (hamishwillee) [inline files/en-us/glossary/fetch_metadata_request_header/index.html:21]: This is a good doc on the headers, and how you use them to create a resource isolation policy. It omits the User header. We could do something similar, but link is appropriate IMO for now. I have linked this and the following playground in all the docs.

**Comment by hamishwillee** [comment]
This is ready for review. @annevk @Elchi3 You may be interested in providing feedback.

File: files/en-us/glossary/fetch_metadata_request_header/index.html (hunk 3 above)
Comment (annevk) [reply files/en-us/glossary/fetch_metadata_request_header/index.html:21]: It seems better to link this (and the one below) after the list of the headers in its own paragraph.

File: files/en-us/glossary/http_header/index.html [hunk 4]
```diff
@@ -8,12 +8,13 @@
...
 
-<p>The current HTTP specifications refer to a number of header categories, including:</p>
+<p>The HTTP and Fetch specifications refer to a number of header categories, including:</p>
 
 <ul>
  <li>{{Glossary("Request header")}}: Headers containing more information about the resource to be fetched or about the client itself.</li>
  <li>{{Glossary("Response header")}}: Headers with additional information about the response, like its location or about the server itself (name, version, …).</li>
  <li>{{Glossary("Representation header")}}: metadata about the resource in the message body (e.g. encoding, media type, etc.).</li>
+ <li>{{Glossary("Fetch metadata request header")}}: Headers wtih metadata about the resource in the message body (e.g. encoding, media type, etc.).</li>
```
Comment (annevk) [inline files/en-us/glossary/http_header/index.html:17]:
```suggestion
 <li>{{Glossary("Fetch metadata request header")}}: Headers with metadata about the resource in the message body (e.g. encoding, media type, etc.).</li>
```

File: files/en-us/web/http/headers/sec-fetch-dest/index.html [hunk 5]
```diff
@@ -132,4 +137,7 @@ <h2 id="See_also">See also</h2>
 	<li>{{HTTPHeader("Sec-Fetch-Mode")}}</li>
 	<li>{{HTTPHeader("Sec-Fetch-Site")}}</li>
 	<li>{{HTTPHeader("Sec-Fetch-User")}}</li>
+	<li><a href="https://web.dev/fetch-metadata/">Protect your resources from web attacks with Fetch Metadata</a> (web.dev)</li> 
+	<li><a href="https://secmetadata.appspot.com/">Fetch Metadata Request Headers playground</a> (secmetadata.appspot.com)</li> 
+
```
Comment (annevk) [inline files/en-us/web/http/headers/sec-fetch-dest/index.html:142]: Nit: spurious newline.

File: files/en-us/web/http/headers/sec-fetch-site/index.html [hunk 6]
```diff
@@ -19,62 +27,58 @@
...
- <dd></dd>
- <dt><code>none</code></dt>
- <dd>This request does not relate to any context like site, origin, or frame. This can happen when user had initiated this request by, e.g. directly entering a URL in the address bar, opening a bookmark, or dragging-and-dropping a file into the browser window.</dd>
-</dl>
+  <dt><code>cross-site</code></dt>
+  <dd>The request initiator and the server hosting the resource have a different site (i.e. a request by "potentially-evil.com" for a resource at "example.com").</dd>
+  <dt><code>same-origin</code></dt>
+  <dd>The request initiator and the server hosting the resource have the same {{Glossary("origin")}} (same scheme, host and port).</dd>
+  <dt><code>same-site</code></dt>
+  <dd>The request initiator and the server hosting the resource have the same domain and/or subdomain, but not necessarily the same scheme and port.</dd>
```
Comment (annevk) [inline files/en-us/web/http/headers/sec-fetch-site/index.html:54]: Do we explain the Public Suffix List somewhere on MDN? Might be worth referencing if we do.

Also, I think same scheme is a requirement here.

**Review by annevk — COMMENTED** [review]
Thanks! Generally looks good to me.

File: files/en-us/web/http/headers/sec-fetch-site/index.html [hunk 7]
```diff
@@ -19,62 +27,58 @@
...
+ </dl>
+
+
+ <h2 id="Examples">Examples</h2>
+
+ <p>A fetch request to <code>https://mysite.example/foo.json</code> originating from a web page on <code>https://mysite.example</code> (with the same port) is a same-origin request.
+The browser will generate the <code>Sec-Fetch-Site: same-origin</code> header as shown below, and the server will typically allow the request:</p>
+
+ <pre class="brush: http">GET /foo.json
+Sec-Fetch-Dest: none
```
Comment (dergoegge) [inline files/en-us/web/http/headers/sec-fetch-site/index.html:66]:
```suggestion
Sec-Fetch-Dest: empty
```

File: files/en-us/web/http/headers/sec-fetch-site/index.html [hunk 8]
```diff
@@ -19,62 +27,58 @@
...
+Sec-Fetch-Mode: cors
+Sec-Fetch-Site: same-origin
+</pre>
 
-<h2 id="Examples">Examples</h2>
+<p>A fetch request to the same URL from another site, for example <code>potentially-evil.com</code>, causes the browser to generate a different header (e.g. <code>Sec-Fetch-Site: cross-site</code>), which the server can choose to accept or reject:</p>
 
-<p>TODO</p>
+<pre class="brush: http">GET /foo.json
+Sec-Fetch-Dest: none
```
Comment (dergoegge) [inline files/en-us/web/http/headers/sec-fetch-site/index.html:74]:
```suggestion
Sec-Fetch-Dest: empty
```

**Review by dergoegge — CHANGES_REQUESTED** [review]
`none` is not a valid value for `sec-fetch-dest`

But other than that, this looks good to me.

File: files/en-us/glossary/fetch_metadata_request_header/index.html (hunk 3 above)
Comment (hamishwillee) [reply files/en-us/glossary/fetch_metadata_request_header/index.html]: Thanks. Good point. I have handled this by pulling the headers under their own bullet "Related headers". Looks a lot better.

File: files/en-us/web/http/headers/sec-fetch-dest/index.html (hunk 5 above)
Comment (hamishwillee) [reply files/en-us/web/http/headers/sec-fetch-dest/index.html]: Fixed thanks.

File: files/en-us/web/http/headers/sec-fetch-site/index.html (hunk 8 above)
Comment (hamishwillee) [reply files/en-us/web/http/headers/sec-fetch-site/index.html:74]: Thanks!

File: files/en-us/web/http/headers/sec-fetch-site/index.html (hunk 6 above)
Comment (hamishwillee) [reply files/en-us/web/http/headers/sec-fetch-site/index.html:54]: @annevk Good point. Unfortunately not, which is IMO an omission. We need something like https://web.dev/same-site-same-origin/. I have created #5557 to track creation, but it should not hold up this PR. 

Regarding "Also, I think same scheme is a requirement here.". My interpretation of the spec was that same site here means "schemelessly same site": https://html.spec.whatwg.org/multipage/origin.html#schemelessly-same-site - can you confirm - happy to change it if you're _sure_ that the scheme is a requirement.

Comment (annevk) [reply files/en-us/web/http/headers/sec-fetch-site/index.html:54]: See https://w3c.github.io/webappsec-fetch-metadata/#abstract-opdef-set-site. It references https://html.spec.whatwg.org/multipage/origin.html#same-site.

Comment (hamishwillee) [reply files/en-us/web/http/headers/sec-fetch-site/index.html:54]: Thanks very much - I'll change it (FWIW that's the link I followed and why I thought the scheme doesn't matter - even with your confirmation I find this non-intuitive).

**Comment by hamishwillee** [comment]
@Rumyra This has been checked by the technical experts, so should be good for your review and merge.

**Review by Rumyra — APPROVED** [review]
Thanks @hamishwillee - awesome work 👍

**Pull request merged**